### DISCUSSION TITLE
How to flash Yeelight lamps without soldering (!)

### DISCUSSION THREAD

**Discussion by helgek** [syssi/esphome-yeelight-ceiling-light#72] [Show and tell]
**Please note:** There's also an "open source" approach (haven't tested this myself) to flash without soldering using a non-commercial approach which works with acupuncture needles. You can find more about this here: https://www.superhouse.tv/36-3d-printed-pcb-workstation-using-acupuncture-needles/

**Update Mar 2024:** There's also this interesting tool but due to difficult accessibility situation with some of the Yeelight's ESP 32 it may not be suitable for some of the lamps: https://www.tindie.com/products/petl/esp32-wroom-pogo-pin-programmer-board/
The term to search for is: "pogo pins"

**My approach / success:** I managed to flash my two Arwen 550S, one Arwen 450C and one meteorite (Crystal Pendant Light) without soldering after I found an outstanding tool online which could be a game changer for many of you who aren't soldering champions but still desperately want to flash a stable firmware on your Yeelight lamps.

**The solution is called PCBite.** Some genius engineers from this Swedish company have invented this amazing tool. You can find their web presence at: https://sensepeek.com/

With this tool even a soldering moron like me can accomplish the flashing exercise which seemed to be a goal out of reach especially when I faced the PCBs of the Arwen lamps.

The Arwen 450C had an additional challenge ready for me since you have to cut the plastic cover to actually be able to reach inside; and then the locking mechanism revealed that this lamp was designed this way that there's no way to open it again except by cutting into the cover of the lamp; in addition the PCB was very difficult to hold for the flashing exercise since it wasn't as flexible to be moved like it was the case with my two Arwen 550S.

I have no clue how people soldered e.g. the Arwen 450C since it's seems impossible to reach some of the pins for soldering due to how they're squeezed between two PCBs. Even with PCBite it was still an artistic exercise to place some of the probes to hold on the pins of the microcontroller. But luckily it worked out in the end for all of my Yeelight lamps 🥳 

Now with 4 ESPhome flashed yeelight lamps I can report that they now all work 100 percent reliable, no connection drops, no disappearing from the network anymore (especially the original meteorite firmware is terrible), simply no more issues at all! And it has been like this for several months now (I had flashed the lamps about 3 months ago)!

For everyone who's interested, **I ordered these two PCBite sets**:

Sensepeek 4005 SP10
Sensepeek 6003 PCBite

In addition I had bought helping arms, like these ones: https://www.amazon.de/gp/product/B09V7SZ6XB/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

And for those who need the UART-TTL USB adapter, I had flawless experience with this one (in connection with a Mac device): https://www.amazon.de/gp/product/B0B7RHPMT7/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

Of course this is quite an investment if you own only one lamp, but even for me it took some consideration to actually go this path. I didn't want to return the PCBite after I had flashed my lamps since this would just have been not fair towards this company. Considering the great quality of the product and its uniqueness I think it's still very fairly priced. For me it was clear that their was no alternative to the yeelight lamps in my apartment, so I eventually decided to do it, and gladly I can afford it.

Thank you again @syssi  and everyone else who contributed to this project! 

Here are two photos showing my successful flashing efforts with PCBite (Meteorite/Crystal Pendant Light and Arwen 450C):

![IMG_3632](https://github.com/syssi/esphome-yeelight-ceiling-light/assets/6280991/ae0082c8-a29a-4842-8fb9-a34d5b9644ee)

![266170393-ede1506b-a6b7-4084-9305-9a2285c9e106](https://github.com/syssi/esphome-yeelight-ceiling-light/assets/6280991/a136c714-a631-4bc4-b725-58d340fd688b)

**Comment by helgek** [comment]
@syssi As discussed please find my summary in this thread which you maybe want to feature in the tutorial.

**Comment by rdorsch** [comment]
For the second photograph (I assume Arwen 450C or 550C) you could also use the test points as described [here](https://github.com/syssi/esphome-yeelight-ceiling-light/issues/22#issuecomment-1133876977)

**Comment by helgek** [comment]
@rdorsch Yes indeed the second photo was the 450C (the "worst" one in regards of reachability), the first photo shows the Meteorite lamp. Thank you for the additional info!

**Reply by PsychoRS** [reply]
Hi @helgek ,

Are those "reachability" problems found also on the Arwen 450S/550S or only in the 450C?

Thanks

**Comment by helgek** [comment]
@PsychoRS I also experienced it with the 550S, only the frequency was different, maybe only every few weeks/months. Now for more than one year my lamps were always working

**Comment by PsychoRS** [comment]
550S flashed! No soldering thanks to the PCBite tools, thanks for the recommendation @helgek!!

![IMG20250111160652](https://github.com/user-attachments/assets/60215e62-e2f0-4b2a-b873-1f2d633e40ec)
![IMG20250111160705](https://github.com/user-attachments/assets/0c8dfad1-efdf-484a-8506-b76c7aadb7c2)

**Reply by helgek** [reply]
Congrats @PsychoRS !!! The 550S is already one of the more challenging ones, I was also extremely happy when I successfully flashed it!! I have two 550S and one 450C and one meteorite pendant light in my apartment

**Reply by PsychoRS** [reply]
> Congrats @PsychoRS !!! The 550S is already one of the more challenging ones, I was also extremely happy when I successfully flashed it!! I have two 550S and one 450C and one meteorite pendant light in my apartment

IDK how I would flash the 550S without PCBite tools LOL.

I have the 550S, a Meteorite Pendant Light and a Bedside Lamp 2, all running ESPHome thanks to the info of this @syssi repo :D
